### PR TITLE
Convert tests in test/presenters to rspec

### PR DESCRIPTION
**Pull request by sihugh**
I've left `require_relative "../../test/support/coronavirus_helper"` where it is because it's used by other tests that haven't been moved yet.

I've converted some stubs to doubles.

I've removed a small amount of duplication where it looked simple.

The diff in the second commit is more faithful. Github seems to get confused about file moves.

Part of https://github.com/alphagov/collections/issues/2259

https://trello.com/c/EzMBK0uf/1426-convert-services-tests-to-rspec

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/presenters/announcements_presenter_spec.rb [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-    end
+    let(:presenter) { described_class.new("boris-johnson.cy", filter_key: "people") }
 
     it "uses a locale-less slug in the links" do
-      links = @presenter.links
-      assert_equal links[:email_signup], "/email-signup?link=/government/people/boris-johnson"
-      assert_equal links[:subscribe_to_feed], "/search/news-and-communications.atom?people=boris-johnson"
-      assert_equal links[:link_to_news_and_communications], "/search/news-and-communications?people=boris-johnson"
+      links = presenter.links
+      expect("/email-signup?link=/government/people/boris-johnson").to eq(links.[](:email_signup))
```
Comment (hannako) [inline spec/presenters/announcements_presenter_spec.rb:7]: `.[]` in a few places 😬

File: spec/presenters/changed_documents_pagination_presenter_spec.rb [hunk 2]
```diff
@@ -19,137 +15,134 @@ def build_presenter_for_subtopic(total: 100, start: 0, page_size: 50, view_conte
...
+      expect(presenter.total_pages).to eq(0)
     end
   end
 
   describe "#next_page?" do
     it "is true if there is a subsequent page" do
       presenter = build_presenter_for_subtopic(start: 0, total: 100)
 
-      assert presenter.next_page?
+      expect(presenter).to be_next_page
```
Comment (hannako) [inline spec/presenters/changed_documents_pagination_presenter_spec.rb:38]: I've been manually changing these sorts of things to eg 
`expect(presenter.next_page).to be(true)`

Comment (hannako) [reply spec/presenters/changed_documents_pagination_presenter_spec.rb:38]: like you have in the next test actually - you must just have missed this one

Comment (hannako) [reply spec/presenters/changed_documents_pagination_presenter_spec.rb:38]: same comment for lines 52 and 58 below

File: spec/presenters/organisations_presenter_spec.rb [hunk 3]
```diff
@@ -196,46 +177,48 @@
...
-      content_item = ContentItem.new(ministerial_departments_hash)
-      content_store_organisations = ContentStoreOrganisations.new(content_item)
-      @organisations_presenter = Organisations::IndexPresenter.new(content_store_organisations)
-    end
+    let(:organisations_presenter) { presenter_from_hash(ministerial_departments_hash) }
 
     it "executive_office? returns true if organisation is number_10" do
       test_organisation_type = :number_10
-      assert @organisations_presenter.executive_office?(test_organisation_type)
+      expect(organisations_presenter.executive_office?(test_organisation_type)).to be
```
Comment (hannako) [inline spec/presenters/organisations_presenter_spec.rb:189]: one of the weird `.to be`'s slipped through your net

File: spec/presenters/organisations_presenter_spec.rb [hunk 4]
```diff
@@ -196,46 +177,48 @@
...
     end
 
     it "executive_office? returns false if organisation is not number_10" do
       test_organisation_type_ministerial = :ministerial_departments
       test_organisation_type_non_ministerial = :public_corporations
 
-      assert_not @organisations_presenter.executive_office?(test_organisation_type_ministerial)
-      assert_not @organisations_presenter.executive_office?(test_organisation_type_non_ministerial)
+      expect(organisations_presenter.executive_office?(test_organisation_type_ministerial)).to_not be true
+      expect(organisations_presenter.executive_office?(test_organisation_type_non_ministerial)).to_not be true
```
Comment (hannako) [inline spec/presenters/organisations_presenter_spec.rb:197]: `.to be(false)`?

File: spec/presenters/organisations_presenter_spec.rb (hunk 3 above)
Comment (hannako) [reply spec/presenters/organisations_presenter_spec.rb:189]: same comment 203, 209

File: spec/presenters/organisations_presenter_spec.rb (hunk 4 above)
Comment (hannako) [reply spec/presenters/organisations_presenter_spec.rb:197]: same 215

File: spec/presenters/changed_documents_pagination_presenter_spec.rb (hunk 2 above)
Comment (sihugh) [reply spec/presenters/changed_documents_pagination_presenter_spec.rb]: Ta. They're quite hard to see after a while 👍

File: spec/presenters/organisations_presenter_spec.rb (hunk 3 above)
Comment (sihugh) [reply spec/presenters/organisations_presenter_spec.rb]: 👍

File: spec/presenters/organisations_presenter_spec.rb (hunk 4 above)
Comment (sihugh) [reply spec/presenters/organisations_presenter_spec.rb]: I was feeling whimsical. Thanks :-)

File: spec/presenters/announcements_presenter_spec.rb (hunk 1 above)
Comment (sihugh) [reply spec/presenters/announcements_presenter_spec.rb]: I found a couple more in an old model spec, so have added those to this commit too

File: spec/models/list_set_spec.rb [hunk 5]
```diff
@@ -114,7 +114,7 @@
     end
 
     it "includes the base_path for all items" do
-      expect(list_set_empty_data.first.contents.to_a.[](3).base_path).to eq("/pay-paye-tax")
+      expect(list_set_empty_data.first.contents.to_a[3].base_path).to eq("/pay-paye-tax")
     end
```
Comment (hannako) [inline spec/models/list_set_spec.rb:118]: thanks :)

**Review by hannako — APPROVED** [review]

**Pull request merged**